src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs

```diff
@@ -180,6 +183,14 @@ public override SyntaxNode VisitGroupClause(GroupClauseSyntax node)
                         continue;
                     }
 
+                    if (fieldName[0] == '(' && fieldName.Trim(')').EndsWith("null"))
+                    {
```

Comment:
`group by item.Gnull` ?

Reply:
Not sure what you mean. 
- `Gnull` is valid field name identifier
- `group by item.null` isn't valid c# code

Reply:
`group by item.@null` is , and will be compiled as `item.null`, no?

Reply:
oh you meant `@null`, but this case isn't grouping by constant. then it's properly of document / map result